server/src/main/java/org/elasticsearch/health/node/LocalHealthMonitor.java

```diff
@@ -112,42 +135,80 @@ private void maybeScheduleNextRun(TimeValue time) {
     }
 
     // Helper method that starts the monitoring without a delay.
-    private void maybeScheduleNow() {
+    // Visible for testing
+    void maybeScheduleNow() {
         maybeScheduleNextRun(TimeValue.ZERO);
     }
 
     @Override
     public void clusterChanged(ClusterChangedEvent event) {
-        if (prerequisitesFulfilled == false) {
-            prerequisitesFulfilled = event.state().nodesIfRecovered().getMinNodeVersion().onOrAfter(Version.V_8_5_0)
-                && HealthMetadata.getFromClusterState(event.state()) != null;
-            maybeScheduleNow();
+        DiscoveryNode current = HealthNode.findHealthNode(event.state());
+        prerequisitesFulfilled = event.state().nodesIfRecovered().getMinNodeVersion().onOrAfter(Version.V_8_5_0)
+            && HealthMetadata.getFromClusterState(event.state()) != null
+            && current != null;
+        if (prerequisitesFulfilled) {
+            DiscoveryNode previous = HealthNode.findHealthNode(event.previousState());
+            if (Objects.equals(previous, current) == false) {
```

Comment:
We've discussed this in another channel, but this check doesn't work because at least sometimes `HealthMetadata.getFromClusterState(event.state())` becomes non-null in a ClusterChangedEvent after the ClusterChangedEvent where the health node changes. In that case `previous` will always be equal to `current`, and so the two variables below don't get reset, so the health updates are never sent to the new health node.

Reply:
Nice find @masseyke , I was planning to rework this code to be more readable and now I will include your suggestion to compare the `currentHealthNode` with the `lastSeenHealthNode`. That should improve things considerably!